modules/rate-sampling/src/main/scala/io/janstenpickle/trace4cats/rate/sampling/RateSpanSampler.scala

```diff
@@ -1,34 +1,54 @@
 package io.janstenpickle.trace4cats.rate.sampling
 
-import cats.Functor
+import cats.Applicative
 import cats.effect.kernel.{Resource, Temporal}
 import cats.syntax.functor._
 import io.janstenpickle.trace4cats.kernel.SpanSampler
 import io.janstenpickle.trace4cats.model.{SampleDecision, SpanContext, SpanKind, TraceId}
-import io.janstenpickle.trace4cats.rate.TokenBucket
+import io.janstenpickle.trace4cats.rate.{TokenBucket, TokenInterval}
 
 import scala.concurrent.duration._
 
 object RateSpanSampler {
-  def apply[F[_]: Functor: TokenBucket]: SpanSampler[F] =
+  def apply[F[_]: Applicative: TokenBucket]: SpanSampler[F] = apply(rootSpansOnly = true)
+
+  def apply[F[_]: Applicative: TokenBucket](rootSpansOnly: Boolean): SpanSampler[F] =
     new SpanSampler[F] {
       override def shouldSample(
         parentContext: Option[SpanContext],
         traceId: TraceId,
         spanName: String,
         spanKind: SpanKind
-      ): F[SampleDecision] =
-        TokenBucket[F].request1.map {
+      ): F[SampleDecision] = {
+        val sampleDecisionF: F[SampleDecision] = TokenBucket[F].request1.map {
           case false => SampleDecision.Drop
           case true => SampleDecision.Include
         }
+
+        parentContext.map(_.traceFlags.sampled).fold(sampleDecisionF) {
+          case SampleDecision.Include =>
+            if (rootSpansOnly) Applicative[F].pure(SampleDecision.Include)
+            else sampleDecisionF
+          case SampleDecision.Drop => Applicative[F].pure(SampleDecision.Drop)
+        }
+      }
+
```

Comment:
I've change the functionality here to be align more with the probabilistic sampler. We may want to backport.

Reply:
Sure. Let's backport as much as we can if it's easy.